packages/in_app_purchase/lib/src/store_kit_wrappers/sk_product_wrapper.dart

```diff
@@ -4,9 +4,52 @@
 
 import 'package:flutter/foundation.dart';
 
+/// Response type for a [startProductRequest] request.
+///
+/// Contains information about a list of products and a list of invalid product identifers.
+/// this is a wrapper for StoreKit's [SKProductsResponse](https://developer.apple.com/documentation/storekit/skproductsresponse?language=objc).
+class SkProductResponseWrapper {
+  SkProductResponseWrapper(
+      {@required this.products, @required this.invalidProductIdentifiers});
+
+  /// constructor to build with a map
+  ///
+  /// Used for constructing the class with map passed from the OBJC layer.
+  SkProductResponseWrapper.fromMap(Map<String, List<dynamic>> map)
+      : products = _getListFromMapList(_getProductMapListFromResponseMap(map)),
+        invalidProductIdentifiers =
+            List.castFrom<dynamic, String>(map['invalidProductIdentifiers']);
+
+  /// The list of the products.
+  ///
+  /// Can be null if the [SKProductRequestMaker]'s [startProductRequest] method does not pass correct [productIdentifiers].
+  final List<SKProductWrapper> products;
+
+  /// The list of invalid product identifier.
+  ///
+  /// If any of the product identifer in [productIdentifers] is invalid, it will be stored here. It can be null if all the product identifiers are valid.
+  final List<String> invalidProductIdentifiers;
+
+  static List<Map<dynamic, dynamic>> _getProductMapListFromResponseMap(
+      Map<String, List<dynamic>> map) {
+    return map['products'].cast<Map<dynamic, dynamic>>();
+  }
+
+  static List<SKProductWrapper> _getListFromMapList(
+      List<Map<dynamic, dynamic>> mapList) {
+    return mapList
+        .map((Map<dynamic, dynamic> map) => SKProductWrapper.fromMap(map))
+        .toList();
+  }
+}
+
 /// A unit discription the length of a subscription period.
 ///
-/// This is wrapper of StoreKit's [SKProductPeriodUnit](https://developer.apple.com/documentation/storekit/skproductperiodunit?language=objc).
+/// This is wrapper for StoreKit's
+/// [SKProductPeriodUnit](https://developer.apple.com/documentation/storekit/skproductperiodunit?language=objc).
+/// This is used as a property in the [SKProductSubscriptionPeriodWrapper].
+/// The values of the enum options are matching the [SKProductPeriodUnit]'s values. Should there be an update or addition
```

Comment:
I don't think this bit about `SKProductPeriodUnit` being kept in sync is public information for the users of the API, it's more of an implementation note for developers on the API. Can you move it out of the Dart Doc into a private comment? (This also applies to `ProductDiscountPaymentMode`)